solr/core/src/test/org/apache/solr/handler/TestReplicationHandler.java

```diff
@@ -261,23 +259,22 @@ private NamedList<Object> getIndexVersion(SolrClient s) throws Exception {
     return res;
   }
 
-  private NamedList<Object> reloadCore(SolrClient s, String core) throws Exception {
+  private void reloadCore(JettySolrRunner jettySolrRunner, String core) throws Exception {
 
     ModifiableSolrParams params = new ModifiableSolrParams();
     params.set("action", "reload");
     params.set("core", core);
     params.set("qt", "/admin/cores");
     QueryRequest req = new QueryRequest(params);
 
-    try (SolrClient adminClient = adminClient(s)) {
+    try (SolrClient adminClient = adminClient(jettySolrRunner)) {
       NamedList<Object> res = adminClient.request(req);
       assertNotNull("null response from server", res);
-      return res;
     }
   }
 
-  private SolrClient adminClient(SolrClient client) {
-    String adminUrl = ((HttpSolrClient) client).getBaseURL().replace("/collection1", "");
+  private SolrClient adminClient(JettySolrRunner client) {
+    String adminUrl = client.getBaseUrl().toString().replace("/collection1", "");
     return getHttpSolrClient(adminUrl);
   }
```

Comment:
This method demonstrates a problem that has bothered me with Solr's Http clients for a long time.  The user can construct an HttpSolrClient with a URL that includes the collection, but HttpSolrClient doesn't "know" this.  This has consequences, such as the existence of this method and issues elsewhere.  Ideally, the "base URL" would be provided and optionally a default collection/core name.  Then, here, we wouldn't need another SolrClient instance simply because we want to issue admin commands.  In a test setting, we could even avoid Jetty/HTTP.  Granted, addressing that is definitely out of scope.  

For now, without trying to fix all imperfections at once, the change here is fine.

Reply:
Do you want to create a JIRA and associate it with the right other jira's to tackle this in the future?

Reply:
https://issues.apache.org/jira/browse/SOLR-16563